### PR TITLE
Improve Homepage, ObjectChange Performance

### PR DESCRIPTION
**Pull request by bryanculver**
# Closes: #3410
# What's Changed

- Add -time index for ObjectChange because we typically lookup records by newest first
- Use cached user name on record instead of str(User)
- Stop joining User table to ObjectChange records for the homepage query of 15 latest changes

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [NA] Attached Screenshots, Payload Example
- [NA] Unit, Integration Tests
- [NA] Documentation Updates (when adding/changing features)
- [NA] Example Plugin Updates (when adding/changing features)
- [NA] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/templates/extras/inc/panel_changelog.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
                 {% endif %}
                 <br />
                 <small>
-                    <span class="text-muted">{{ change.user|default:change.user_name }} -</span>
+                    <span class="text-muted">{{ change.user_name }} -</span>
```
Comment (bryanculver) [inline nautobot/extras/templates/extras/inc/panel_changelog.html:23]: The odds that the latest 15 ObjectChange records are from a user who changed their username are?

**Review by gsnider2195 — APPROVED** [review]

File: nautobot/extras/homepage.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def get_job_results(request):
 
 def get_changelog(request):
     """Callback function to collect changelog for panel."""
-    return ObjectChange.objects.restrict(request.user, "view").select_related("user", "changed_object_type")[:15]
+    return ObjectChange.objects.restrict(request.user, "view").select_related("changed_object_type")[:15]
```
Comment (glennmatthews) [inline nautobot/extras/homepage.py:17]: I'm really curious how much of a performance improvement this change makes. The added index I can absolutely see making a huge difference but is one join really a significant issue?

**Review by glennmatthews — APPROVED** [review]

**Review by HanlinMiao — APPROVED** [review]
🚀

File: nautobot/extras/homepage.py (hunk 2 above)
Comment (bryanculver) [reply nautobot/extras/homepage.py:17]: When @glennmatthews was pulling the `explain query` it was for some reason doing a sequential scan of the database.

File: nautobot/extras/templates/extras/inc/panel_changelog.html (hunk 1 above)
Comment (jathanism) [reply nautobot/extras/templates/extras/inc/panel_changelog.html:23]: -0

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**